compiler/z/codegen/OMRMachine.cpp

```diff
@@ -3007,7 +3007,7 @@ OMR::Z::Machine::findBestFreeRegister(TR::Instruction   *currentInstruction,
       preference = 0;
       }
 
-   uint64_t prefRegMask = _registerFile[preference] ? _registerFile[preference]->getRealRegisterMask() : 0;
+   uint64_t prefRegMask = TR::RealRegister::isRealReg((TR::RealRegister::RegNum)preference) ? _registerFile[preference]->getRealRegisterMask() : 0;
```

Comment:
Just to elaborate a bit more on my concern, I also see a few other array accesses of `_registerFile` in this same function a little further below. These also seem scary to me as `preference` could clearly be a non-real register. There are also numerous other array accesses of `_registerFile` in other functions in the codegen and I'm not certain if we are exceeding bounds on those as well.

This is enough cause for concern that I think we should properly size these arrays and `NULL` initialize values which we deem not valid (all non-real registers).

Reply:
I don't believe an invalid array access is possible inside this function. The above line will set `prefRegMask` to 0 if `preference` is a non-real register. After this line, in every case where we try to access `_registerFile[preference]` we first check if `prefRegMask` is not 0. 

I agree with the concern though. This could be a problem in the future if new code is added and  proper boundary checking is not done.

Reply:
Yes, you're correct. All the paths are guarded indeed. I glossed over this very important detail. Thanks!